src/products/controller/productCategories.controller.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  NotFoundException,
+  Param,
+  Post,
+  Put,
+} from '@nestjs/common';
+import ProductCategoriesMapper from '../mapper/productCategory.mapper';
+import { ProductCategoriesService } from '../service/productCategories.service';
+import ProductCategory from '../domain/productCategory.domain';
+import ProductCategoryDTO from '../dto/productCatergory.dto';
+import { ProductsService } from '../service/products.service';
+import { ApiResponse } from '@nestjs/swagger';
+
+@Controller('productCategories')
+export class ProductCategoriesController {
+  constructor(
+    private productCategoriesMapper: ProductCategoriesMapper,
+    private productCategoriesService: ProductCategoriesService,
+    private productsService: ProductsService,
+  ) {}
+
+  @ApiResponse({ status: 200, description: 'Returns the product categories' })
+  @Get()
+  async getProducts(): Promise<ProductCategory[]> {
+    return await this.productCategoriesService.getProductCategories();
+  }
+
+  @ApiResponse({ status: 200, description: 'Returns the product category' })
+  @ApiResponse({
+    status: 404,
+    description: "The product category can't be found",
+  })
+  @Get(':id')
+  async getProductById(
+    @Param() { id }: { id: string },
+  ): Promise<ProductCategory | null> {
```

Comment:
Controllers should also return DTOs, not entities. It's good to decouple the database internal implementation from the response required by clients (even if for now the entity is identical to the required response).

Reply:
done, every controller's every endpoint returns dto